bin/cjson-parser.c

```diff
@@ -0,0 +1,106 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <cjson/cJSON.h>
+
+void readJSONFile(const char *filename);
+void iterJSONFile(cJSON *json);
+
+int main() {
+    const char *filename = "config.json";
+    readJSONFile(filename);
+    return 0;
+}
+
+void iterJSONFile(cJSON *json) {
+ 
+    cJSON *proc_key = cJSON_GetObjectItemCaseSensitive(json, "process");
```

Comment:
I’d recommend either a `cJSON_GetObject` based approach or an iterative approach, but not a mixed approach like you currently have now.

I wonder also if you could have a table-based approach of some kind that maps keys to functions for a given element. E.g. (you’d need some supporting types and the nesting won’t work):

```
callback_map = [ [ "ociVersion", cj_ociVersion ],
                 [ "process", [ "env", cj_env ] ] ]
```

Notably, the `cJSON` type is singly linked and doesn’t include pointers to parent, which would be useful.